Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+        }
+
+        public override void OnStopServer()
+        {
+            lobbySlots.Clear();
+            base.OnStopServer();
+        }
+
+        public override void OnStopHost()
+        {
+            OnLobbyStopHost();
+        }
+
+        #endregion
+
+        #region client handlers
+
+        public override void OnStartClient(NetworkClient lobbyClient)
+        {
+
+            if (lobbyPlayerPrefab == null || lobbyPlayerPrefab.gameObject == null)
+                Debug.LogError("NetworkLobbyManager no LobbyPlayer prefab is registered. Please add a LobbyPlayer prefab.");
+            else
+                ClientScene.RegisterPrefab(lobbyPlayerPrefab.gameObject);
+
+            if (playerPrefab == null)
+                Debug.LogError("NetworkLobbyManager no GamePlayer prefab is registered. Please add a GamePlayer prefab.");
+            else
+                ClientScene.RegisterPrefab(playerPrefab);
+
+            OnLobbyStartClient(lobbyClient);
+        }
+
+        public override void OnClientConnect(NetworkConnection conn)
+        {
+            OnLobbyClientConnect(conn);
+            CallOnClientEnterLobby();
+            base.OnClientConnect(conn);
+        }
+
+        public override void OnClientDisconnect(NetworkConnection conn)
+        {
+            OnLobbyClientDisconnect(conn);
+            base.OnClientDisconnect(conn);
+        }
+
+        public override void OnStopClient()
+        {
+            OnLobbyStopClient();
+            CallOnClientExitLobby();
+
+            if (!string.IsNullOrEmpty(offlineScene))
+            {
+                // Move the LobbyManager from the virtual DontDestroyOnLoad scene to the Game scene.
+                // This let's it be destroyed when client changes to the Offline scene.
+                SceneManager.MoveGameObjectToScene(gameObject, SceneManager.GetActiveScene());
+            }
+        }
+
+        public override void OnClientChangeScene(string newSceneName)
+        {
+            if (LogFilter.Debug) Debug.LogFormat("OnClientChangeScene from {0} to {1}", SceneManager.GetActiveScene().name, newSceneName);
+
+            if (SceneManager.GetActiveScene().name == LobbyScene && newSceneName == GameplayScene && dontDestroyOnLoad && IsClientConnected() && client != null)
+            {
+                GameObject lobbyPlayer = client?.connection?.playerController?.gameObject;
+                if (lobbyPlayer != null)
+                {
+                    lobbyPlayer.transform.SetParent(null);
+                    DontDestroyOnLoad(lobbyPlayer);
+                }
+                else
+                    Debug.LogWarningFormat("OnClientChangeScene: lobbyPlayer is null");
+            }
+            else
+               if (LogFilter.Debug) Debug.LogFormat("OnClientChangeScene {0} {1} {2}", dontDestroyOnLoad, IsClientConnected(), client != null);
+        }
+
+        public override void OnClientSceneChanged(NetworkConnection conn)
```

Comment:
is it 100% necessary to use multiple scenes for lobby?
Mirror doesn't like that too much. why not keep it all in one scene?

Reply:
My example works perfectly with Offline, Lobby, and Gameplay scenes.  Mirror is flakey on its own, but I have it working here.